source/event_stream.c

```diff
@@ -0,0 +1,1118 @@
+    aws_write_u64(header.header_value.static_val, (uint64_t)value);
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int aws_event_stream_add_uuid_header(struct aws_array_list *headers, const char *name, uint8_t name_len,
+                                     const uint8_t *value) {
+    struct aws_event_stream_header_value_pair header = {
+            .header_name_len = name_len,
+            .header_value_len = 16,
+            .value_owned = 0,
+            .header_value_type = AWS_EVENT_STREAM_HEADER_UUID,
+    };
+
+    memcpy((void *) header.header_name, (void *) name, (size_t) name_len);
+    memcpy((void *) header.header_value.static_val, value, 16);
+
+    return aws_array_list_push_back(headers, (void *) &header);
+}
+
+int8_t aws_event_stream_header_value_as_byte(struct aws_event_stream_header_value_pair *header) {
+    return (int8_t)header->header_value.static_val[0];
+
+}
+
+const char *aws_event_stream_header_value_as_string(struct aws_event_stream_header_value_pair *header) {
+    return (const char *) header->header_value.variable_len_val;
+}
+
+int8_t aws_event_stream_header_value_as_bool(struct aws_event_stream_header_value_pair *header) {
+    return header->header_value_type == AWS_EVENT_STREAM_HEADER_BOOL_TRUE ? (int8_t) 1 : (int8_t) 0;
+}
+
+int16_t aws_event_stream_header_value_as_int16(struct aws_event_stream_header_value_pair *header) {
+    return (int16_t)aws_read_u16(header->header_value.static_val);
+}
+
+int32_t aws_event_stream_header_value_as_int32(struct aws_event_stream_header_value_pair *header) {
+    return (int32_t)aws_read_u32(header->header_value.static_val);
+}
+
+int64_t aws_event_stream_header_value_as_int64(struct aws_event_stream_header_value_pair *header) {
+    return (int64_t)aws_read_u64(header->header_value.static_val);
+}
+
+const uint8_t *aws_event_stream_header_value_as_bytebuf(struct aws_event_stream_header_value_pair *header) {
+    return header->header_value.variable_len_val;
+}
+
+int64_t aws_event_stream_header_value_as_timestamp(struct aws_event_stream_header_value_pair *header) {
+    return aws_event_stream_header_value_as_int64(header);
+}
+
+const uint8_t *aws_event_stream_header_value_as_uuid(struct aws_event_stream_header_value_pair *header) {
+    return header->header_value.static_val;
+}
+
+uint16_t aws_event_stream_header_value_length(struct aws_event_stream_header_value_pair *header) {
+    return header->header_value_len;
+}
+
+static struct aws_event_stream_message_prelude empty_prelude = {
+        .total_len = 0,
+        .headers_len = 0,
+        .prelude_crc = 0
+};
+
+static void reset_state(struct aws_event_stream_streaming_decoder *decoder);
+
+static void reset_header_state(struct aws_event_stream_streaming_decoder *decoder, uint8_t free_header_data);
+
+static int
+headers_state(struct aws_event_stream_streaming_decoder *decoder, const uint8_t *data, size_t len, size_t *processed);
```

Comment:
inconsistent style

Reply:
what's inconsistent here? that's the style output from clang-format

Reply:
updated regardless